src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/FormLoginFactory.php

```diff
@@ -48,8 +48,27 @@ public function addConfiguration(NodeDefinition $node)
         parent::addConfiguration($node);
 
         $node
+            ->beforeNormalization()
+                ->ifTrue(function ($v) { return isset($v['csrf_provider']) && isset($v['csrf_token_generator']); })
+                ->thenInvalid("You should define a value for only one of 'csrf_provider' and 'csrf_token_generator' on a security firewall. Use 'csrf_token_generator' as this replaces 'csrf_provider'.")
+            ->end()
+            ->beforeNormalization()
+                ->ifTrue(function ($v) { return isset($v['intention']) && isset($v['csrf_token_id']); })
+                ->thenInvalid("You should define a value for only one of 'intention' and 'csrf_token_id' on a security firewall. Use 'csrf_token_id' as this replaces 'intention'.")
+            ->end()
+            ->beforeNormalization()
+                ->ifTrue(function ($v) { return isset($v['csrf_provider']); })
+                ->then(function ($v) {
+                    @trigger_error("Setting the 'csrf_provider' configuration key on a security firewall is deprecated since version 2.8 and will be removed in 3.0. Use the 'csrf_token_generator' configuration key instead.", E_USER_DEPRECATED);
+
+                    $v['csrf_token_generator'] = $v['csrf_provider'];
+                    unset($v['csrf_provider']);
+
+                    return $v;
+                })
+                ->end()
             ->children()
-                ->scalarNode('csrf_provider')->cannotBeEmpty()->end()
+                ->scalarNode('csrf_token_generator')->cannotBeEmpty()->end()
             ->end()
```

Comment:
I removed the beforeNormalizer for "intention": there is not "intention" in this configuration section